src/middlewared/middlewared/plugins/ftp.py

```diff
@@ -9,6 +9,16 @@ class FTPService(SystemServiceService):
     class Config:
         service = "ftp"
         datastore_prefix = "ftp_"
+        datastore_extend = "ftp.ftp_extend"
+
+    async def ftp_extend(self, data):
+        if data['ssltls_certificate']:
```

Comment:
It has been discussed we should not have nested foreign key in in services query/config. It has been a bad idea to begin with because model does not necessarily match the query/config which leads to us having to manually do more calls like these. We should probably just fix it to return the id and not the object.

Yes, there are a number of places doing this wrong but we can slowly change them later.